src/queue/descriptor_chain.rs

```diff
@@ -178,3 +185,838 @@ impl<M: GuestAddressSpace> Iterator for DescriptorChainRwIter<M> {
         }
     }
 }
+
+#[derive(Clone, Debug)]
+struct DescriptorChainConsumer<'a> {
+    buffers: VecDeque<VolatileSlice<'a>>,
```

Comment:
I didn't write the code. Your arguments makes a lot of sense, what makes you think it couldn't be improved after the API is introduced? I would rather land something that works now, and improve later. Especially because I didn't write the code in the first place, and we have users of this code, so it works well enough.

Reply:
As you suggested, I switched the code to use a Vec. Using an iterator for VolatileSlice seems a lot more involved, I would like to postpone it.

Reply:
I'm mainly trying to understand if some the approaches/methods that exist right now can be changed going forward, or they are actually required by certain use cases. In a sense, consuming bytes from a descriptor chain feels a bit more natural using stream operations vs random access ones, and this makes me wonder if (for example) methods like `split_at` and `available_bytes` can be removed/replaced in the future. Just trying to ensure I have the right model(s) in mind :D

Reply:
split_at() seems to be there for convenience, it could probably be removed. But I think internal implementation details could also allow a combined implementation, using a VolatileSlice iterator or a Vec, depending on the usage. This fall in performance details to me, that can be addressed later.